### ISSUE TITLE
Discover 2.0 - Revamping Discover to be able to be used for all of OpenSearch Dashboards data exploration use cases

### ISSUE THREAD

**Issue by elfisher** [opensearch-project/OpenSearch-Dashboards#3056]
This is a place holder for a proposal to revamp Discover so that it can be used in all of our experiences that require data exploration rather than having separate document exploration user experiences. For example today Event Explorer and Discover have similar functionality and are used for similar use cases. This proposal is to revamp discover to work across all these use cases.

A more detailed proposal will follow.

**Comment by joshuarrrr** [comment]
> For example today Event Explorer and Discover have similar functionality and are used for similar use cases.

I know this is just a placeholder, but it's absolutely critical that we're able to actually articulate the current use cases of Discover and Event Explorer, as well as specific features/functionality, before we can assess where the overlaps and gaps are and choose a technical approach.

How is this issue distinct from https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1792? Should we update #1792 to be an RFC that asks for specific use cases or pain points from the community?

**Comment by joshuarrrr** [comment]
I've also tagged discover-related bugs and feature requests with the https://github.com/opensearch-project/OpenSearch-Dashboards/labels/discover label, so that it's easier to track existing known issues.

**Comment by elfisher** [comment]
@joshuarrrr you are right. This duplicative/similar to #1792.

**Comment by ananzh** [comment]
Add two draft Issues in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1792 ([here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1792#issuecomment-1358930527)) and also update the same contents below
* Understand Discover which includes setup & start discover and core functions  https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3112
* Rewrite Evaluation: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3104

**Comment by BSFishy** [comment]
@ahopp to provide guidance when the working backwards doc for Discover 2.0 is completed

**Comment by ashwin-pc** [comment]
Proposal for Data Explorer/Discover 2.0 outlined in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4165

**Comment by zengyan-amazon** [comment]
Superseded by #4165.
